app/controllers/Extractors.scala

```diff
@@ -80,6 +84,61 @@ class Extractors  @Inject() (extractions: ExtractionService,
     }
   }
 
+  def showExtractorMetrics(extractorName: String) = AuthenticatedAction { implicit request =>
+    implicit val user = request.user
+
+    val dateFormatter = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSS")
+    val todaydate = dateFormatter.format(new java.util.Date())
+
+    val hourcal = Calendar.getInstance
+    hourcal.add(Calendar.HOUR, -24)
+    val yesterdaydate = dateFormatter.format(hourcal.getTime)
+
+    val daycal = Calendar.getInstance
+    daycal.add(Calendar.WEEK_OF_MONTH, -1)
+    val last7daydate = dateFormatter.format(daycal.getTime)
+
+    val monthcal = Calendar.getInstance
+    monthcal.add(Calendar.MONTH, -1)
+    val lastmonthdate = dateFormatter.format(monthcal.getTime)
+
+    Logger.debug("today date: " + todaydate)
+    Logger.debug("yesterday date: " + yesterdaydate)
+    Logger.debug("last 7 day date: " + last7daydate)
+    Logger.debug("last month date: " + lastmonthdate)
+
+    val myDonelist = extractions.findByExtractorIDBefore(extractorName, "DONE", yesterdaydate, 10)
```

Comment:
Can you add some simple comments to explain what the code is doing? Just enough to make it easier to read. Thank you.

Reply:
fixed.